torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -190,11 +205,23 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
             self.to_sharded_dtensor(padded_sharded_param[: sharded_param.size(0)])
         )
         self.sharded_param.requires_grad_(param.requires_grad)
-        unsafe_free_storage(param_data)  # free immediately
-        del param_data  # delete PyObject reference to avoid warning
+        # Let `param_data` be freed normally when its ref count reaches 0 when
+        # the `fully_shard` call returns to allow provided parameters to alias
         self._setattr_on_modules(self.sharded_param)
         self.sharded_state = ShardedState.SHARDED
 
+    def _init_sharded_post_forward_param_metadata(self, param: torch.Tensor) -> None:
+        mesh_info = self.post_forward_mesh_info
+        assert mesh_info is not None  # mypy
+        param_data = param._local_tensor if isinstance(param, DTensor) else param
```

Comment:
In the future, this can also be `distribute_tensor`. The same two to-dos as `_init_sharded_param()` apply.